src/ogd/games/BLOOM/features/GameCompletionStatus.py

```diff
@@ -0,0 +1,34 @@
+# import libraries
+from typing import Any, List, Optional
+from ogd.core.generators.Generator import GeneratorParameters
+from ogd.core.generators.extractors.Feature import Feature
+from ogd.core.models.Event import Event
+from ogd.core.models.enums.ExtractionMode import ExtractionMode
+from ogd.core.models.FeatureData import FeatureData
+
+class GameCompletionStatus(Feature):
+    def __init__(self, params: GeneratorParameters):
+        super().__init__(params=params)
+        self.status: Optional[str] = "IN_PROGRESS"
+
+    # *** IMPLEMENT ABSTRACT FUNCTIONS ***
+    @classmethod
+    def _eventFilter(cls, mode: ExtractionMode) -> List[str]:
+        return ["win_game", "lose_game"]
+
+    @classmethod
+    def _featureFilter(cls, mode: ExtractionMode) -> List[str]:
+        return []
+
+    def _updateFromEvent(self, event: Event) -> None:
+        if event.EventName == "win_game":
+            self.status = "WIN"
+        elif event.EventName == "lose_game":
+            if self.status != "WIN":
```

Comment:
We can remove the inner `if self.status != "WIN":` line - if a player somehow manages to lose after winning, then we still want to update the status to "LOSS".

Reply:
resolved